controllers/networkfence_controller.go

```diff
@@ -58,5 +155,78 @@ func (r *NetworkFenceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 func (r *NetworkFenceReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&csiaddonsv1alpha1.NetworkFence{}).
+		WithEventFilter(predicate.GenerationChangedPredicate{}).
 		Complete(r)
 }
+
+// validateNetworkFenceSpec validates the NetworkFence spec and checks if values are neither nil nor empty.
+func validateNetworkFenceSpec(nwFence *csiaddonsv1alpha1.NetworkFence) error {
+	if nwFence.Spec.Driver == "" {
+		return errors.New("required parameter 'driver' is not specified")
+	}
+	if nwFence.Spec.Secret.Name == "" {
+		return errors.New("required parameter `secret.name` is not specified")
+	}
+	if nwFence.Spec.Secret.Namespace == "" {
```

Comment:
these 2  ( name or namespace) can be clubbed as only one does not help.

Reply:
The secret is a struct value, we can't compare it directly. I feel we have to compare values only. Feel free to suggest changes if you feel otherwise